onnxruntime/core/util/thread_utils.cc

```diff
@@ -17,19 +17,25 @@ static std::unique_ptr<ThreadPool>
 CreateThreadPoolHelper(Env* env, OrtThreadPoolParams options) {
   if (options.thread_pool_size == 1)
     return nullptr;
-  std::vector<size_t> cpu_list;
   ThreadOptions to;
+
   if (options.affinity_vec_len != 0) {
-    to.affinity.assign(options.affinity_vec, options.affinity_vec + options.affinity_vec_len);
+    to.affinity.reserve(options.affinity_vec_len);
+    std::transform(options.affinity_vec, options.affinity_vec + options.affinity_vec_len, std::back_inserter(to.affinity),
+                   [](size_t affinity) {
+                     return ThreadOptions::ThreadAffinity{{static_cast<int>(affinity)}};
```

Comment:
>return ThreadOptions::ThreadAffinity{{static_cast<int>(affinity)}}; [](http://example.com/codeflow?start=21&length=67)

maybe we need some comment here - each affinity from options.affinity_vec represent an ordinal id of the processor, not a bitmask of processors.

Reply:
another note here - in near future we might need to refactor this since this approach prohibit the ability of setting multiple per-thread affinity with a integer.
e.g. for 1st thread we want attach its to processor 1,2,3,4

Reply:
1) Would not commenting at the header definition site a better idea?

Reply:
Will comment